### PR TITLE
Last.fm API

### PR DESCRIPTION
**Pull request by Darlite**

### CODE REVIEW

File: lastFm-api/js/request.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        console.error(this.statusText);
+      }
+    }
+  };
+  this.xhr.onerror = function () {
+    console.error(this.statusText);
+  };
+}
+
+Request.prototype.sendRequest = function () {
```
Comment (WebTownIP) [inline lastFm-api/js/request.js]: what about to pass request params in `senRequest`?

Comment (Darlite) [reply lastFm-api/js/request.js]: I changed the `sendRequest()` function, so now it can work with other methods too.

Comment (NikitaSedyx) [reply lastFm-api/js/request.js]: hm, I mean to pass param object {author: 'Char', ...}

Comment (Darlite) [reply lastFm-api/js/request.js]: I don't understand, why do we need this object?

File: lastFm-api/js/request.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Request.prototype.makeGet = function (queryParams) {
+  queryParams.api_key = this.apiKey;
+  queryParams.format = this.format;
+  var queryString = this.url + buildQueryString(queryParams);
+  this.xhr.open('GET', queryString, true);
+  this.xhr.send(null);
+};
+
+function buildQueryString (queryParams) {
```
Comment (NikitaSedyx) [inline lastFm-api/js/request.js]: I'd suggest to include these functions to prototype

**Review by NikitaSedyx — APPROVED** [review]
there is an advice, but it's 100% :+1:

File: lastFm-api/js/request.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  };
+  this.xhr.onerror = function () {
+    console.error(this.statusText);
+  };
+}
+
+Request.prototype.makeGet = function (queryParams) {
+  queryParams.api_key = this.apiKey;
+  queryParams.format = this.format;
+  var queryString = this.url + buildQueryString(queryParams);
```
Comment (NikitaSedyx) [inline lastFm-api/js/request.js]: don't forget to update this line

File: lastFm-api/js/request.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  this.xhr.open('GET', queryString, true);
+  this.xhr.send(null);
+};
+
+Request.prototype.buildQueryString = function (queryParams) {
+  return encodeQueryData(queryParams);
```
Comment (NikitaSedyx) [inline lastFm-api/js/request.js]: and this one,

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**